pkg/util/vsphere/vm/hardware_version.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package vm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+
+	"github.com/vmware-tanzu/vm-operator/pkg/util"
+)
+
+const HardwareVersionProperty = "summary.config.hwVersion"
+
+func ReconcileMinHardwareVersion(
+	ctx context.Context,
+	client *vim25.Client,
+	mo mo.VirtualMachine,
+	fetchProperties bool,
+	minHardwareVersion int32) (bool, error) {
+
+	var (
+		hwVersion int32
+		log       = logr.FromContextOrDiscard(ctx)
+		obj       = object.NewVirtualMachine(client, mo.Self)
+	)
+
+	log = log.WithValues(
```

Comment:
Can we early return when `minHardwareVersion == 0` b/c that will be the most common value and we know we have nothing to do?

Also, this seems like a lot of repeated logging in the steady state when the VM hardware version will change at most a few times in its lifetime.